### PR TITLE
feat: 1142 orchard step cypress test

### LINKED ISSUE #1142: Orchard Step Cypress Test
Test Step 4 - Orchard with Cypress
Add a new spec: `../a-class-seedlot-reg-form-orchard.cy.ts`

## Information

An orchard has a set of parent trees, e.g. 
- orchard 109 has parent tress [1, 3, 5, 7, 9]
- orchard 110 has parent trees [2, 3, 5, 4, 6, 8, 10]

A user can select up to 2 orchards

If one chard is selected in Orchard step, for example orchard 109, step 5 should display parent tree [1, 3, 5, 7, 9]. 

If two orchards are selected, the union of the two orchards parent trees should be displayed in step 5

For example, a user selects orchard 109 and orchard 110
On step 5, the displayed parents are [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

## Test parent tree union

select ORCHARD_A only
37
69
101 
129 
192

select ORCHARD_B only
37 
38  
53
54 
60

UNION ORCHARD_A AND ORCHARD B 
Get a set of parent tree numbers SET X = [37, 38, 53, 54, 60, 69, 129, 192] 

Select ORCHARD_A and ORCHARD_B
find the first  `X.length()` number of rows' parent numbers

Compare if the two sets are the same

## Other testing

Test the remaining inputs

### PR DESCRIPTION
**Pull request by ngunner15**
# Description
Closes #1142 
Testing for this step
image

#### New
- Cypress test for Orchard step

#### Changed
- cypress.config
- A class seedlot fixture

#### Removed
- None

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [x] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-27-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1227-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)
- [Oracle-API](https://nr-spar-1227-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+      .should('have.value', '');
+
+    // Save changes
+    cy.get('.seedlot-registration-button-row')
+      .find('button.form-action-btn')
+      .contains(/Save changes|Changes saved!/g)
+      .click();
+  });
+
+  it('store second Parent Tree Number in an array', () => {
```
Comment (craigyu) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts:279]: Second orchard's parent tress

Comment (ngunner15) [reply frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts]: https://github.com/bcgov/nr-spar/pull/1227/commits/346e20d74f8e2877ada02c40afaa0cdface299b4

**Review by craigyu — APPROVED** [review]
Good job!

**Review by craigyu — CHANGES_REQUESTED** [review]
image

it's failing

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+      .click();
+
+    cy.get('.gametic-combobox')
+      .eq(0)
+      .find('ul li')
+      .eq(1)
+      .click();
+
+    cy.get('#orchard-female-gametic')
+      .should('have.value', 'F2 - Measured Cone Volume');
```
Comment (craigyu) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts:435]: Also I noticed some things that can be improved on, let's have a discussion tomorrow. 

How can we know for sure that the first gametic-combobox is the female one, and how do we know the second option you click will always be `F2 - Measured Cone Volume`?

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+
+    cy.get('#orchard-combobox-0')
+      .should('have.value', '');
+
+    // Save changes
+    cy.get('.seedlot-registration-button-row')
+      .find('button.form-action-btn')
+      .contains(/Save changes|Changes saved!/g)
+      .click();
+  });
```
Comment (craigyu) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts:218]: need to update this method, use ownership as a reference

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts (hunk 2 above)
Comment (ngunner15) [reply frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts]: https://github.com/bcgov/nr-spar/pull/1227/commits/a24c2984605e6699c7f5f8e2fe137caac8704941

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts [hunk 4]
```diff
@@ -0,0 +1,607 @@
...
+      .should('have.value', '');
+
+    cy.get('#controlled-cross-no')
+      .should('be.checked');
+
+    cy.get('#biotech-no')
+      .should('be.checked');
+  });
+
+  it.only('Change gamete information', () => {
```
Comment (craigyu) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts:437]: mistake

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts [hunk 5]
```diff
@@ -0,0 +1,607 @@
...
+      .siblings()
+      .click();
+
+      cy.get(`.${prefix}--list-box--expanded`)
+      .find('ul li')
+      .eq(2)
+      .click();
+
+    cy.get('#orchard-male-gametic')
+      .should('have.value', 'M3 - Pollen Volume Estimate by 100% Survey');
```
Comment (craigyu) [inline frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts:462]: how is this different than your previous implementation? 
it's still clicking on an item in a list with an arbitrary position number, then checking if has the correct value.
As we discussed before, we shouldn't assume an item will be at a specific position in a list dynamically generated with backend data

What needs to be done instead:

1. open the combobox
2. find a specific element, for example, find the M3 in the list, then click on it
3. Verify M3 has been selected

**Review by craigyu — CHANGES_REQUESTED** [review]

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts (hunk 4 above)
Comment (ngunner15) [reply frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts]: https://github.com/bcgov/nr-spar/pull/1227/commits/874c8cdb080b3cbc62ae2e97dc0754134133b5ec

File: frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts (hunk 5 above)
Comment (ngunner15) [reply frontend/cypress/e2e/smoke-test/a-class-seedlot-reg-form-orchard.cy.ts]: https://github.com/bcgov/nr-spar/pull/1227/commits/874c8cdb080b3cbc62ae2e97dc0754134133b5ec

**Review by RMCampos — APPROVED** [review]
LGTM

**Comment by craigyu** [comment]
#1329 has been merged, can you replace the save functions here? +1 🍺 virtual beer for effort

**Comment by DerekRoberts** [comment]
Two and three weeks is a long time to keep PRs open, mostly due to sitting on resources.  Please reopen this and #1273 when needed.

**Review by craigyu — APPROVED** [review]

**Pull request merged**